<launch>
    <include file="$(find gazebo_ros)/launch/empty_world.launch">
      <arg name="paused" value="true"/>
      <arg name="use_sim_time" value="true"/>
      <arg name="gui" value="true"/>
      <arg name="headless" value="false"/> 		<!-- Inert - see gazebo_ros_pkgs issue #491 -->
      <arg name="recording" value="false"/>
      <arg name="debug" value="false"/>
    </include>

    <!-- Spawn a robot into Gazebo -->
    <param name="robot_description" command="$(find xacro)/xacro --inorder $(find toy_problem)/urdf/vbmbot.xacro"/>
    <node name="urdf_spawner" pkg="gazebo_ros" type="spawn_model" respawn="false" output="screen"
    args="-urdf -model vbmbot -param robot_description
      -J joint1 0
      -J joint2 0
      -unpause" />

    <!-- <node name="spawn_robot_env" pkg="gazebo_ros" type="spawn_model" args="-file $(find toy_problem)/urdf/robot_env.urdf -urdf -x -0.25 -z 0 -model robot_env" /> -->
    <node name="spawn_robot_env" pkg="gazebo_ros" type="spawn_model" args="-file $(find toy_problem)/urdf/robot_env.urdf -urdf -x -0.1 -z 0 -model robot_env" />
    
    <!-- Launch controller for robot joints -->
    <rosparam file="$(find toy_problem)/config/config.yaml" command="load"/>

    <!-- load the controllers -->
    <node name="controller_spawner_stopped" pkg="controller_manager" type="spawner" respawn="false" output="screen" ns="/vbmbot" args=
      "--stopped joint1_velocity_controller
      		 joint2_velocity_controller"/>

    <!-- load & start the position controllers -->
    <node name="controller_spawner" pkg="controller_manager" type="spawner" respawn="false" output="screen" ns="/vbmbot" args=
      "joint_state_controller
       joint1_position_controller
       joint2_position_controller"/>

    <!-- convert joint states to TF transforms -->
    <node pkg="robot_state_publisher" type="robot_state_publisher" name="rob_st_pub" respawn="false" output="screen">
    	<remap from="/joint_states" to="/vbmbot/joint_states" />
    </node> 
    
    <!-- <node name="controller_node" pkg="toy_problem" type="vbmbot_joint_controller.py" output="screen" /> -->
    

</launch>
